nixops/__main__.py

```diff
@@ -527,6 +527,16 @@
     help="include the physical specification in the evaluation",
 )
 
+subparser = add_subparser(
+    subparsers, "eval", help="eval the given file is nix code in the network expression"
+)
+subparser.set_defaults(op=op_eval)
+subparser.add_argument("code", metavar="CODE", help="code")
+subparser.add_argument(
+    "--json", action="store_true", help="print the option value in JSON format"
+)
+subparser.add_argument("--strict", action="store_true", help="enable strict evaluation")
```

Comment:
I don't have a specific suggestion, but we might want to crib what the nix-instantiate manual says:


> recursively evaluate list elements and attributes. Normally, such sub-expressions are left unevaluated
> (since the Nix expression language is lazy).
>    Warning
>    This option can cause non-termination, because lazy data structures can be infinitely large.

I wonder why someone wouldn't use `--strict --json`? Should `--json` imply `--strict`?

Reply:
Yes, `--json` will only work with `--strict` if it's more than one level deep.

The use-case here was if you wanted to evaluate something but finely control the level of strictness and output Nix instead of JSON. For that you'd want lazy evaluation of functions, which is impossible if `--strict` is hardcoded.

Given that this is supposed to be used by other tooling (that's why I don't pass a plain string but a file instead), it shouldn't be too confusing or hard to configure it for the specific usage.